all/test.sh

```diff
@@ -56,6 +64,15 @@ testGeneration() {
         fi
     fi
 
+    if [[ "$lang" == "java" ]]; then
+        if [[ "$extra_args" == *"-o gen/test.jar" ]]; then
```

Comment:
Not sure this works if `extra_args` is truly an array and the option and its argument are two separate elements in the array.

So maybe you want just a scalar, in which case line 26 should probably be `extra_args="$@"`. (But Codacy might still complain about the quoteless expansion.)

Or, use an array, and change the comparison here to work with that.

Reply:
OR or, convert the array to a string temporarily here and compare that, which might be easiest?

Reply:
I followed the same pattern used throughout this file. I don't think it makes sense to do something differently here than all the other tests

Reply:
Agreed!